h2o-py/tests/testdir_algos/gam/pyunit_PUBDEV_8877_gam_beta_constraints.py

```diff
@@ -0,0 +1,39 @@
+from __future__ import division
+from __future__ import print_function
+from past.utils import old_div
+import sys
+sys.path.insert(1, "../../../")
+import h2o
+from tests import pyunit_utils
+from h2o.estimators.gam import H2OGeneralizedAdditiveEstimator
+
+# In this test, we check and make sure splines_non_negative should take precedence over the non_negative parameter
+def tea_gam_non_negative_spline_non_negative():
+    train1 = h2o.import_file(pyunit_utils.locate("smalldata/gam_test/nondecreasingCosTest.csv"))
+    gamX = ["a"]
+    x = ["a"]
+    numKnots = [4]
+    scale= [0.001]
+    bs_type = [2]
+    splines_non_negative = [False]
+    spline_order = [3]
+    # build GAM with non_negative set to False  The I-spline specification should take precedence
+    h2o_model = H2OGeneralizedAdditiveEstimator(gam_columns=gamX, scale=scale, bs=bs_type, spline_orders=spline_order,
+                                                num_knots=numKnots,  seed=12345, non_negative = True, 
+                                                splines_non_negative=splines_non_negative)
+    h2o_model.train(x = [], y="cosy", training_frame=train1)
+    # check that all coefficients are positive except for the ones for I-spline that is set to be non-positive
+    coefs = h2o_model.coef()
+    del coefs["Intercept"]
+    non_positive_names = ["a_is_0", "a_is_1", "a_is_2", "a_is_3", "a_is_4"]
+    for key in coefs.keys():
```

Comment:
When I run it, the ouput for coef is {'a_is_0': 0.0, 'a_is_1': 0.0, 'a_is_2': 0.0, 'a_is_3': 0.0, 'a_is_4': 0.0, 'Intercept': -0.15340742509662778}

My understanding is that the 0 will always pass no matter what the original problem was. The test passed for latest rel-zygmund also.

Reply:
I made changes to the test.  This is the result I get: 
![image](https://user-images.githubusercontent.com/7231712/201982101-9edd9e6b-c05a-4a9d-a2e0-760f17f1b859.png)